orangecontrib/text/widgets/owpubmed.py

```diff
@@ -0,0 +1,463 @@
+                'Retrieve records',
+                callback=self.retrieve_records,
+                tooltip='Performs a search for articles that fit the '
+                        'specified parameters.')
+        self.pubmed_controls.append(self.retrieve_records_button)
+
+        # Num. retrieved records info label.
+        self.retrieval_info_label = gui.label(
+                self.controlArea,
+                self,
+                'Number of records retrieved: /')
+
+        # Load the most recent emails.
+        self.set_email_list()
+
+        # Load the most recent queries.
+        self.set_keyword_list()
+
+        # Check the email and enable controls accordingly.
+        if self.recent_emails:
+            email = self.recent_emails[0]
+            self.email_is_valid = validate_email(email)
+
+        self.enable_controls()
+
+    def sync_email(self):
+        email = self.email_combo.currentText()
+        self.email_is_valid = validate_email(email)
+        self.enable_controls()
+
+    def enable_controls(self):
+        # Enable/disable controls accordingly.
+        for control in self.pubmed_controls:
+            control.setEnabled(self.email_is_valid)
+        if self.pubmed_api is None or self.pubmed_api.search_record_count == 0:
+            self.retrieve_records_button.setEnabled(False)
+        if not self.email_is_valid:
+            self.email_combo.setFocus()
+
+    def run_search(self):
+        self.warning(0)
+        self.run_search_button.setEnabled(False)
+        self.retrieve_records_button.setEnabled(False)
+
+        # Add the email to history.
+        email = self.email_combo.currentText()
+        if email not in self.recent_emails:
+            self.recent_emails.insert(0, email)
+
+        # Check if the PubMed object is present.
+        if self.pubmed_api is None:
+            self.pubmed_api = Pubmed(
+                    email=email,
+                    progress_callback=self.api_progress_callback,
+                    error_callback=self.api_error_callback,
+            )
+
+        if self.search_tabs.currentIndex() == 0:
+            # Get query parameters.
+            terms = self.keyword_combo.currentText().split()
+            authors = self.author_input.text().split()
+
+            # If no keywords, alert that the query is too vague.
+            if not terms:
+                self.warning(0, 'Please specify the keywords for this query.')
+                self.run_search_button.setEnabled(True)
+                self.retrieve_records_button.setEnabled(True)
+                return
+
+            # Check date formatting.
+            pdate_from_flag = self.pub_date_from and not eval_date(
+                    self.pub_date_from
+            )
+            pdate_to_flag = self.pub_date_to and not eval_date(
+                    self.pub_date_to
+            )
+            if pdate_from_flag or pdate_to_flag:
+                self.warning(0, 'Please specify dates with digits in '
```

Comment:
When you use [a correct widget](https://github.com/biolab/orange3-text/pull/17/files#r59973329), this is a non-problem.